### ISSUE TITLE
feeding Low Res images to Generator without downsampling High Res images

### ISSUE THREAD

**Issue by mustaeenqazi** [HasnainRaz/Fast-SRGAN#24]
Hi Hasnain,
Thanks for the great code, i have triained it and tested on custom images and it works very well and it is very computationally efficient compared to other SRGAN models i have used,
For my project i need to feed the Generator with low resolution images separately (from a separate directory), without downsampling the high res images. How should i change the file Dataloader.py 
any help would be appreciated,
i tried something and i have a code snippet as an example of how to load low res images using a generator function but it gives an error.
![loading images separately](https://user-images.githubusercontent.com/66418760/98856557-04252100-245e-11eb-8bd3-90c48447d6cf.JPG)

**Comment by HasnainRaz** [comment]
You should modify "dataloader.py" to accept two image_dirs (low res and high res).
something like:

```python
class DataLoader(object):

    def __init__(self, low_res_image_dir, high_res_image_dir):
        self.low_res_image_paths = [os.path.join(low_res_image_dir, x) for x in os.listdir(low_res_image_dir)]
        self.high_res_image_paths = [os.path.join(high_res_image_dir, x) for x in os.listdir(high_res_image_dir)]
        # Not sure if your images in both dirs have the save names and are paired. If so, sorting will make both
        # lists 1:1 correspondent.
    def _parse_image(self, image_path):
        image = tf.io.read_file(image_path)
        image = tf.image.decode_jpeg(image, channels=3)
        image = tf.image.convert_image_dtype(image, tf.float32)

        return image

   def _parse_images(self, low_res_path, high_res_path):
       low_res_image = self._parse_image(low_res_path)
       high_res_image = self._parse_image(high_res_path)

       return low_res_image, high_res_image

   def dataset(self, batch_size, threads=4):
        """
        Returns a tf dataset object with specified mappings.
        Args:
            batch_size: Int, The number of elements in a batch returned by the dataset.
            threads: Int, CPU threads to use for multi-threaded operation.
        Returns:
            dataset: A tf dataset object.
        """

        # Generate tf dataset from image paths.
        dataset = tf.data.Dataset.from_tensor_slices((self.low_res_image_paths, self.high_res_image_paths))

        # Read the both images
        dataset = dataset.map(self._parse_images, num_parallel_calls=tf.data.experimental.AUTOTUNE)

        # Batch the input, drop remainder to get a defined batch size.
        # Prefetch the data for optimal GPU utilization.
        dataset = dataset.shuffle(30).batch(batch_size, drop_remainder=True).prefetch(tf.data.experimental.AUTOTUNE)

        return dataset
```

This will not work probably, as there is no crop or resize function, so this code will fail to batch data. But hopefully you get the idea. You can then add the required cropping/resizing and normalizing functions to this code.
